### PR TITLE
Convert CallHandler to typescript

### PR DESCRIPTION
**Pull request by dbkr**
and remove the old conference call stuff while we're at it: enough
time should have passed since those mistakes that we can move on.
The old conference call rooms will still appear for anyone whose
account dates back to that time, but they've presumably been appearing
in any other matrix client they used too.

Requires https://github.com/vector-im/element-web/pull/15274

### CODE REVIEW

File: src/@types/global.d.ts [hunk 1]
```diff
@@ -62,6 +64,7 @@ declare global {
 
     interface Navigator {
         userLanguage?: string;
+        mediaSession: any;
```
Comment (t3chguy) [inline src/@types/global.d.ts:67]:
```suggestion
        // https://github.com/Microsoft/TypeScript/issues/19473
        // https://developer.mozilla.org/en-US/docs/Web/API/MediaSession
        mediaSession: any;
```

File: src/CallHandler.tsx [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+import {WidgetType} from "./widgets/WidgetType";
+import {SettingLevel} from "./settings/SettingLevel";
+import {base32} from "rfc4648";
+
+import QuestionDialog from "./components/views/dialogs/QuestionDialog";
+import ErrorDialog from "./components/views/dialogs/ErrorDialog";
+
+export default class CallHandler {
+    private calls = {};
+    private audioPromises = {};
```
Comment (t3chguy) [inline src/CallHandler.tsx:78]: type?

File: src/CallHandler.tsx [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+        for (let i = 0; i < roomsWithCalls.length; i++) {
+            if (this.calls[roomsWithCalls[i]] &&
+                    this.calls[roomsWithCalls[i]].call_state !== "ended") {
+                return this.calls[roomsWithCalls[i]];
+            }
+        }
+        return null;
+    }
+
+    play(audioId) {
```
Comment (t3chguy) [inline src/CallHandler.tsx:118]: type

File: src/CallHandler.tsx [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+                    audio.load();
+                    return playAudio();
+                });
+            } else {
+                this.audioPromises[audioId] = playAudio();
+            }
+        }
+    }
+
+    pause(audioId) {
```
Comment (t3chguy) [inline src/CallHandler.tsx:147]: type

File: src/CallHandler.tsx [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+            if (this.audioPromises[audioId]) {
+                this.audioPromises[audioId] = this.audioPromises[audioId].then(() => audio.pause());
+            } else {
+                // pause doesn't actually return a promise, but might as well do this for symmetry with play();
+                this.audioPromises[audioId] = audio.pause();
+            }
+        }
+    }
+
+    private setCallListeners(call) {
```
Comment (t3chguy) [inline src/CallHandler.tsx:161]: what type is this?

File: src/CallHandler.tsx [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+                this.setCallState(call, call.roomId, "stop_ringing");
+                this.pause("ringbackAudio");
+            } else if (newState === "connected") {
+                this.setCallState(call, call.roomId, "connected");
+                this.pause("ringbackAudio");
+            }
+        });
+    }
+
+    private setCallState(call, roomId, status) {
```
Comment (t3chguy) [inline src/CallHandler.tsx:217]: can these get types?

File: src/CallHandler.tsx [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+            button: _t('Try using turn.matrix.org'),
+            cancelButton: _t('OK'),
+            onFinished: (allow) => {
+                SettingsStore.setValue("fallbackICEServerAllowed", null, SettingLevel.DEVICE, allow);
+                cli.setFallbackICEServerAllowed(allow);
+            },
+        }, null, true);
+    }
+
+    private onAction = (payload) => {
```
Comment (t3chguy) [inline src/CallHandler.tsx:268]:
```suggestion
    private onAction = (payload: ActionPayload) => {
```
Will need the import

File: src/CallHandler.tsx [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+                this.setCallState(this.calls[payload.room_id], payload.room_id, "connected");
+                dis.dispatch({
+                    action: "view_room",
+                    room_id: payload.room_id,
+                });
+                break;
+        }
+    }
+
+    private async startCallApp(roomId, type) {
```
Comment (t3chguy) [inline src/CallHandler.tsx:390]: types please

File: src/CallHandler.tsx (hunk 7 above)
Comment (dbkr) [reply src/CallHandler.tsx]: done with import

File: src/CallHandler.tsx (hunk 8 above)
Comment (dbkr) [reply src/CallHandler.tsx]: done

File: src/CallHandler.tsx (hunk 6 above)
Comment (dbkr) [reply src/CallHandler.tsx]: yes they can!

File: src/CallHandler.tsx (hunk 5 above)
Comment (dbkr) [reply src/CallHandler.tsx]: it's a call object from js-sdk which I've just given an alias for any until we ts-ify the js-sdk bits

File: src/CallHandler.tsx (hunk 4 above)
Comment (dbkr) [reply src/CallHandler.tsx]: done

File: src/CallHandler.tsx (hunk 3 above)
Comment (dbkr) [reply src/CallHandler.tsx]: done

File: src/CallHandler.tsx (hunk 2 above)
Comment (dbkr) [reply src/CallHandler.tsx]: done

**Comment by dbkr** [comment]
e2e tests failing because I didn't name the element-web branch the same thing

**Review by t3chguy — APPROVED** [review]

**Pull request merged**